packages/storybook/src/stories/grid-error.stories.tsx

```diff
@@ -0,0 +1,262 @@
+  };
+
+  return (
+    <div className="grid-container">
+      <XGrid rows={rows} columns={cols} />
+    </div>
+  );
+};
+
+export const showErrorApi = () => {
+  const api = useApiRef();
+  const rows = React.useMemo(() => getRows(), []);
+  const cols = React.useMemo(() => getColumns(), []);
+
+  React.useEffect(() => {
+    if (api && api.current) {
+      api.current!.showError({ message: 'Error loading rows!' });
+    }
+  }, [api]);
+
+  return (
+    <div className="grid-container">
+      <XGrid rows={rows} columns={cols} apiRef={api} />
+    </div>
+  );
+};
+export const errorProp = () => {
+  const rows = React.useMemo(() => getRows(), []);
+  const cols = React.useMemo(() => getColumns(), []);
+
+  return (
+    <div className="grid-container">
+      <XGrid rows={rows} columns={cols} error={{ message: 'Error can also be set in props!' }} />
+    </div>
+  );
+};
+function CustomErrorOverlay(props) {
+  return (
+    <GridOverlay className="custom-overlay">
+      <div style={{ textAlign: 'center' }}>
+        <h1>{props.title}</h1>
+        <p>{typeof props.error === 'string' ? props.error : props.error.message}</p>
+      </div>
+    </GridOverlay>
+  );
+}
+
+export const CustomError = () => {
+  const api = useApiRef();
+  const rows = React.useMemo(() => getRows(), []);
+  const cols = React.useMemo(() => getColumns(), []);
+
+  React.useEffect(() => {
+    if (api && api.current) {
+      api.current!.showError({ error: 'Something bad happened!', title: 'BIG ERROR' });
+    }
+  }, [api]);
+
+  return (
+    <div className="grid-container">
+      <XGrid
+        rows={rows}
+        columns={cols}
+        apiRef={api}
+        components={{
+          errorOverlay: CustomErrorOverlay,
+        }}
+      />
+    </div>
+  );
+};
+export const CustomErrorWithException = () => {
+  const rows = React.useMemo(() => getRows(), []);
+  const cols = React.useMemo(() => getColumns(), []);
+  cols[1].cellClassRules = {
+    common: ({ value }) => {
+      if (value === 'alice') {
+        throw new Error('Alice created an error!');
+      }
+      return true;
+    },
+  };
+
+  return (
+    <div className="grid-container">
+      <XGrid
+        rows={rows}
+        columns={cols}
+        components={{
+          errorOverlay: CustomErrorOverlay,
+        }}
+      />
+    </div>
+  );
+};
+function fetchError(): Promise<any> {
+  return new Promise<any>((resolve, reject) => {
+    setTimeout(() => {
+      reject(new Error('Bad fetch'));
+    }, 500);
+  });
+}
+
+export const AsyncErrorApi = () => {
+  const api = useApiRef();
+  const rows = React.useMemo(() => getRows(), []);
+  const cols = React.useMemo(() => getColumns(), []);
+
+  React.useEffect(() => {
+    fetchError().catch((err) => {
+      api.current!.showError(err);
+    });
+  }, [api]);
+
+  return (
+    <div className="grid-container">
+      <XGrid rows={rows} columns={cols} apiRef={api} />
+    </div>
+  );
+};
+
+export const onErrorHandler = () => {
```

Comment:
Need a capital first letter to fix the `react-hooks/rules-of-hooks` warning (soon to be error)

```suggestion
export const OnErrorHandler = () => {
```